core/src/main/scala/kafka/server/FetchSession.scala

```diff
@@ -144,6 +144,10 @@ class CachedPartition(val topic: String,
       if (updateResponseData)
         localLogStartOffset = respData.logStartOffset
     }
+    if (respData.preferredReadReplica.isPresent) {
```

Comment:
If the preferred read replica is the leader, do we need to respond immediately?

Reply:
Is that what this does? I thought `mustRespond` just made sure to include that partition in the response.

Reply:
I think you're right, but that just changes the question. If the preferred read replica is the leader, do we need to include it in the response? Wouldn't that imply that we always have to include the partition in the response?

Reply:
If the leader picks itself as the preferred read replica, we could exclude it from the response (return none from ReplicaSelector). This actually makes sense b/c we are treating the presence of a preferred read replica ID as a signal to "go somewhere else" to the client and parts of the broker response logic. WDYT?